tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/ESEmailAddressContactSearchEngine.java

```diff
@@ -42,12 +45,17 @@ public ESEmailAddressContactSearchEngine(ReactorElasticSearchClient client) {
 
     @Override
     public Publisher<BoxedUnit> index(AccountId accountId, EmailAddressContact contact) {
+        ObjectMapper objMp = new ObjectMapper();
+        String contactJson = "";
+        try {
+            contactJson = objMp.writeValueAsString(contact);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
         String content = String.format("{" +
-                "\"accountId\": \"%s\"," +
-                "\"id\": \"%s\", \"address\": \"%s\"" + "}", accountId.toString(), contact.id().variant(), contact.address());
-
+                "\"accountId\": \"%s\"," + contactJson.substring(1), accountId.toString());
```

Comment:
What is this wizardry?

Please directly generate the right JSON!

Reply:
The field `identifier` of accountId would be generated as it is with Jackson, which is not the expected field in Json. I don't know how to get it done without manually handling.

Reply:
Maybe by using an intermediate DTO POJO to handle serialization of indexed documents as we do for other ElasticSearch usages?

If you have a POJO matching the structure of the indexed document then you could put all the jackson annotations you need in there.

Reply:
Okie, a cumbersome to me but perhaps it's required, will do it!

Reply:
I prefer that to JSON injections for sure...